configs/projects/_shared-agent-settings.rb

```diff
@@ -94,30 +99,32 @@
 platform_triple = "arm-linux-gnueabi" if platform.name == 'debian-8-armel'
 platform_triple = "aarch64-redhat-linux" if platform.name == 'el-7-aarch64'
 
-if platform.is_cross_compiled_linux?
-  host = "--host #{platform_triple}"
-
-  # Use a standalone ruby for cross-compilation
-  proj.setting(:host_ruby, "/opt/pl-build-tools/bin/ruby")
-  proj.setting(:host_gem, "/opt/pl-build-tools/bin/gem")
+if platform.is_windows?
+  proj.setting(:host_ruby, File.join(proj.ruby_bindir, "ruby.exe"))
+  proj.setting(:host_gem, File.join(proj.ruby_bindir, "gem.bat"))
+elsif platform.is_cross_compiled_linux? || (platform.is_solaris? && platform.architecture == 'sparc')
+  # Install a standalone ruby for cross-compiled platforms
+  if platform.name =~ /solaris-10-sparc/
+    proj.setting(:host_ruby, "/opt/csw/bin/ruby")
+    proj.setting(:host_gem, "/opt/csw/bin/gem2.0")
+  else
+    proj.setting(:host_ruby, "/opt/pl-build-tools/bin/ruby")
+    proj.setting(:host_gem, "/opt/pl-build-tools/bin/gem")
+  end
+else
+  proj.setting(:host_ruby, File.join(proj.ruby_bindir, "ruby"))
+  proj.setting(:host_gem, File.join(proj.ruby_bindir, "gem"))
```

Comment:
The logic around solaris and cross-compiled platforms should be unchanged here, although I've reorganized this file a bit. I've just manually compared `inspect` output diffs for 5.5.x and master for solaris (sparc and x86), windows, and several cross-compiled platforms, and everything looks in order to me.